### PR TITLE
Add "velkoja" app for automatically sending notifications of overdue payments

### PR DESCRIPTION
**Pull request by rambo**

### CODE REVIEW

File: project/velkoja/tests.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
+
+# Create your tests here.
```
Comment (tanelikaivola) [inline project/velkoja/tests.py]: Minimum: create a single static NotificationSent entry. (Just make sure a sane object can be saved)

File: project/config/settings/common.py [hunk 2]
```diff
@@ -52,6 +52,7 @@
     'rest_framework',
     'rest_framework.authtoken',
     'django_markdown',
+    'django_jinja',
```
Comment (tanelikaivola) [inline project/config/settings/common.py]: Really sure: "from django.template.loader import render_to_string" isn't enough? Is django_jinja worth the extra dependency?

File: project/velkoja/holvichecker.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            if float(invoice._jsondata.get('credited_sum')) > 0:
+                continue
+            # If we have already sent notification recently, do not sent one just yet
+            if NotificationSent.objects.filter(transaction_unique_id=invoice.code).count():
+                notified = NotificationSent.objects.get(transaction_unique_id=invoice.code)
+                if (timezone.now() - notified.stamp).days < settings.HOLVI_NOTIFICATION_INTERVAL_DAYS:
+                    continue
+
+            if send:
+                invoice.send()
```
Comment (tanelikaivola) [inline project/velkoja/holvichecker.py]: Possible exception here

File: project/velkoja/holvichecker.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            barcode = None
+            if barcode_iban:
+                barcode = bank_barcode(barcode_iban, invoice.rf_reference, Decimal(invoice.due_sum))
+
+            mail = EmailMessage()
+            mail.subject = subject_template.render(Context({ "invoice": invoice, "barcode": barcode })).strip()
+            mail.body = body_template.render(Context({ "invoice": invoice, "barcode": barcode }))
+            mail.to = [invoice.receiver.email]
+            if send:
+                mail.send()
```
Comment (tanelikaivola) [inline project/velkoja/holvichecker.py]: - Possible exception here. What happens if in middle of the process one this fails?
- Didn't we just send an email already at invoice.send()?

File: project/velkoja/holvichecker.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+            try:
+                notified = NotificationSent.objects.get(transaction_unique_id=invoice.code)
+                notified.notification_no += 1
+            except NotificationSent.DoesNotExist:
+                notified = NotificationSent()
+                notified.transaction_unique_id = invoice.code
+            notified.stamp = timezone.now()
+            notified.email = invoice.receiver.email
+            if send:
+                notified.save()
```
Comment (tanelikaivola) [inline project/velkoja/holvichecker.py]: Possible exception here

**Comment by jssmk** [comment]
Add try/except for mail.send() in HolviOverdueInvoicesHandler. Use exception info in NotificationSent to keep track of mail errors.

File: project/velkoja/holvichecker.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            ret.append((notified, invoice))
```
Comment (tanelikaivola) [inline project/velkoja/holvichecker.py]: Should this method somehow handle partial success? Currently it either:
- fully succeeds (no exceptions hit)
- completely fails (without modifying state, hit an exception)
- partially fails (some invoices might have been sent, but this method dies and doesn't return whatever was in ret)

File: docker/dev.sh [hunk 7]
```diff
@@ -1,8 +1,24 @@
-#!/bin/sh
+#!/bin/bash -e
+SCRIPTDIR=$(python -c 'import os,sys;print os.path.dirname(os.path.realpath(sys.argv[1]))' "$0")
```
Comment (rambo) [inline docker/dev.sh]: You do know that this trick is copied verbatim from the scripts you have made ? :)

File: project/config/settings/common.py (hunk 2 above)
Comment (rambo) [inline project/config/settings/common.py]: django_jinja adds some helpers to the templates.

File: project/velkoja/holvichecker.py (hunk 6 above)
Comment (rambo) [inline project/velkoja/holvichecker.py]: So far I'm thinking it's much better to fail hard and have admin fix the actual problem than to try to continue and then report some issue (which will be ignored) somewhere.

File: project/velkoja/tests.py (hunk 1 above)
Comment (rambo) [inline project/velkoja/tests.py]: Minimal saving test has been added.

**Comment by rambo** [comment]
#194 will conflict with the similar fixes here.

**Comment by rambo** [comment]
rebased to `fix_build` ( #194 )

**Pull request merged**